### DISCUSSION TITLE
Can I get current model context within `getUploadedFileNameForStorageUsing()`?

### DISCUSSION THREAD

**Discussion by pekka** [filamentphp/filament#8039] [Help]
I have a `FileUpload` component with a `getUploadedFileNameForStorageUsing()` callback defined as outlined in the docs [here](https://filamentphp.com/docs/3.x/forms/fields/file-upload#generating-custom-file-names).

However, I need the custom file name to contain the slug of the current item that is being edited/created.

Is there a way to get the current item context?

The `$component` argument does appear to contain a `$model` property but it is empty even when I edit an existing item.

**Comment by pboivin** [comment]
Have you tried accessing the field data instead?
```php
->getUploadedFileNameForStorageUsing(function ($get) {
    $slug = $get('slug');
    
    // ...
})
```

**Comment by ikkebra** [comment]
This is how I got it to work

`
                            ->getUploadedFileNameForStorageUsing(
                                fn (TemporaryUploadedFile $file, $record): string => (string) str($record->slug) . '.' . $file->getClientOriginalExtension(),
                            )
`

**Comment by maqndon** [comment]
I made it work like this:

```
->getUploadedFileNameForStorageUsing(function (TemporaryUploadedFile $file, $livewire) {
$slug = ($livewire->data['slug']);
return $slug !== null || $slug !== '' ? (string)str($file->getClientOriginalName())->prepend($slug . '_') : (string)$file->getClientOriginalName();
}),
```
